eduhelx_jupyterlab_prof/handlers.py

```diff
@@ -211,118 +130,259 @@ async def get_value(self, current_path: str):
             # The cwd is the root in the frontend, so treat the path as such.
             # NOTE: IMPORTANT: this field is NOT absolute on the server. It's only the absolute path for the webapp.
             assignment["absolute_directory_path"] = os.path.join("/", rel_assignment_path)
-        value["assignments"] = assignments
 
-        current_assignment = student_repo.current_assignment
-        # The student is in their repo, but we still need to check if they're actually in an assignment directory.
-        if current_assignment is None:
-            # If user is not in an assignment, we're done. Just leave current_assignment as None.
-            return json.dumps(value)
-        
-        submissions = await self.api.get_my_submissions(current_assignment["id"])
-        for submission in submissions:
-            submission["commit"] = get_commit_info(submission["commit_id"], path=student_repo.repo_root)
-        current_assignment["submissions"] = submissions
-        current_assignment["staged_changes"] = []
-        for modified_path in get_modified_paths(path=student_repo.repo_root):
-            full_modified_path = Path(student_repo.repo_root) / modified_path["path"]
-            abs_assn_path = Path(student_repo.repo_root) / assignment["directory_path"]
-            try:
-                path_relative_to_assn = full_modified_path.relative_to(abs_assn_path)
-                modified_path["path_from_repo"] = modified_path["path"]
-                modified_path["path_from_assn"] = str(path_relative_to_assn)
-                current_assignment["staged_changes"].append(modified_path)
-            except ValueError:
-                # This path is not part of the current assignment directory
-                pass
+            assignment["staged_changes"] = []
+            for modified_path in get_modified_paths(path=student_repo.repo_root):
+                full_modified_path = Path(student_repo.repo_root) / modified_path["path"]
+                abs_assn_path = Path(student_repo.repo_root) / assignment["directory_path"]
+                try:
+                    path_relative_to_assn = full_modified_path.relative_to(abs_assn_path)
+                    modified_path["path_from_repo"] = modified_path["path"]
+                    modified_path["path_from_assn"] = str(path_relative_to_assn)
+                    assignment["staged_changes"].append(modified_path)
+                except ValueError:
+                    # This path is not part of the assignment directory
```

Comment:
not worth changing in the PR, but I wonder if down the road we want to log instances of such errors to keep tabs on unintended ways gitea repos are used

Reply:
Sure, we should definitely have logging lol. But we can always just look at students' repos in the gitea UI if we want to know about that.